site/app/controllers/EmailStatusController.php

```diff
@@ -33,4 +32,19 @@ public function getEmailStatusPage() {
             )
         );
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/email/check_announcement", methods={"GET"})
+     * @AccessControl(role="FULL_ACCESS_GRADER")
+     * @return JsonResponse
+     */
+    public function checkAnnouncement() {
+        if (isset($_GET["thread_id"])) {
+            $exists = $this->core->getQueries()->existsAnnouncementsId($_GET["thread_id"]);
+            return JsonResponse::getSuccessResponse([
+                "exists" => $exists
+            ]);
+        }
+        return JsonResponse::getFailResponse("No thread_id provided");
+    }
```

Comment:
An announcement would be a forum post. Due to that I feel that this function should be in the ForumController instead. Or is there a specific reason it is in here?

Reply:
I put this here because this function involves checking with the email table in the database, which doesn't seemed too related to forum to me at the time. But the overall use of this function is very much related to ForumController and to be honest I wasn't sure where I should put this function. What do you think?

Reply:
Typically controller functions should be isolated to where they are used. I see why you were thinking of going with this since it checks the emails but it wouldn't ever be used in the email status page. If someone wanted to go looking for this route it would make sense to find it in the ForumController rather than the EmailStatusController since it is used by the Forum.